pkg/apis/traceflow/v1/types.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Phase uint
+type ComponentType uint
+type ResourceType uint
+type DropReason uint
+type PacketResouceType uint
+type PacketTransportType uint
+
+const (
+	INITIAL Phase = iota
+	RUNNING
+	SUCCESS
+	TIMEOUT
+	ERROR
+)
+
+const (
+	SPOOFGUARD ComponentType = iota
+	LB
+	ROUTING
+	DFW
+	FORWARDING
+)
+
+const (
+	DELIVERED ResourceType = iota
+	RECEIVED
+	FORWARDED
+	DROPPED
+)
+
+const (
+	FieldsPacketData PacketResouceType = iota
+)
+
+const (
+	UNICAST PacketTransportType = iota
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type Traceflow struct {
```

Comment:
Could we resolve all comments and suggestions in the google doc? I saw some definitions do not reflect the latest discussions there.

Reply:
Sure, I am working on it.